tests/run_tests.sh

```diff
@@ -0,0 +1,236 @@
+    echo "    -d             : specify DUT name (*)"
+    echo "    -e             : specify extra parameter(s) (default: none)"
+    echo "    -f             : specify testbed file (default testbed.csv)"
+    echo "    -i             : specify inventory name"
+    echo "    -k             : specify file log level: error|warning|info|debug (default debug)"
+    echo "    -l             : specify cli log level: error|warning|info|debug (default warning)"
+    echo "    -m             : specify test method group|individual|dump (default group)"
+    echo "    -o             : omit the file logs"
+    echo "    -n             : specify testbed name (*)"
+    echo "    -s             : specify list of scripts to skip (default: none)"
+    echo "    -t             : specify toplogy: t0|t1|any|combo like t0,any (*)"
+    echo "    -u             : bypass util group"
+
+    exit $1
+}
+
+function validate_parameters()
+{
+    RET=0
+
+    if [[ -z ${DUT_NAME} ]]; then
+        echo "DUT name (-d) is not set.."
+        RET=1
+    fi
+
+    if [[ -z ${TESTBED_NAME} ]]; then
+        echo "Testbed name (-n) is not set.."
+        RET=2
+    fi
+
+    if [[ -z ${TOPOLOGY} ]]; then
+        echo "TOPOLOGY (-t) is not set.."
+        RET=3
+    fi
+
+    if [[ ${RET} != 0 ]]; then
+        show_help_and_exit ${RET}
+    fi
+}
+
+function setup_environment()
+{
+    SCRIPT=$0
+    FULL_PATH=$(realpath ${SCRIPT})
+    SCRIPT_PATH=$(dirname ${FULL_PATH})
+    BASE_PATH=$(dirname ${SCRIPT_PATH})
+    CLI_LOG_LEVEL='warning'
+    FILE_LOG_LEVEL='debug'
+    SKIP_FOLDERS="ptftests acstests saitests"
+    EXTRA_PARAMETERS=""
+    TEST_CASES=""
+    OMIT_FILE_LOG="False"
+    BYPASS_UTIL="False"
+
+    TEST_METHOD='group'
+
+    TESTBED_FILE="${BASE_PATH}/ansible/testbed.csv"
+    INVENTORY="${BASE_PATH}/ansible/lab,${BASE_PATH}/ansible/veos"
+
+    export ANSIBLE_CONFIG=${BASE_PATH}/ansible
+    export ANSIBLE_LIBRARY=${BASE_PATH}/ansible/library/
+} 
+
+function setup_test_options()
+{
+    PYTEST_COMMON_OPTS="--inventory ${INVENTORY} \
+                      --host-pattern ${DUT_NAME} \
+                      --testbed ${TESTBED_NAME} \
+                      --testbed_file ${TESTBED_FILE} \
+                      --log-cli-level ${CLI_LOG_LEVEL} \
+                      --log-file-level ${FILE_LOG_LEVEL} \
+                      --allow_recover \
+                      --showlocals \
+                      --assert plain \
+                      -rav"
+
+    for skip in ${SKIP_SCRIPTS} ${SKIP_FOLDERS}; do
+        PYTEST_COMMON_OPTS="${PYTEST_COMMON_OPTS} --ignore=${skip}"
+    done
+
+    if [[ x"${OMIT_FILE_LOG}" == x"True" ]]; then
+        UTIL_LOGGING_OPTIONS=""
+        TEST_LOGGING_OPTIONS=""
+    else
+        UTIL_LOGGING_OPTIONS="--junit-xml=logs/util.xml --log-file=logs/util.log"
+        TEST_LOGGING_OPTIONS="--junit-xml=logs/tr.xml --log-file=logs/test.log"
```

Comment:
Would it make sense to have the file name/path be an option? I could see some use in being able to have a `t0.xml` and `any.xml` or something similar if you're chaining multiple test runs together.

Reply:
make sense. let's come back with another PR.